src/api/firebase.js

```diff
@@ -179,22 +179,34 @@ export async function addItem(listPath, { itemName, daysUntilNextPurchase }) {
 	});
 }
 
+/**
+ * Update an item in the user's list in Firestore with new purchase information.
+ * @param {string} listPath The path of the list the item belongs to.
+ * @param {string} itemId The ID of the item being updated.
+ * @param {Object} updatedData Object containing the updated item data.
+ * @param {Date} updatedData.dateLastPurchased The date the item was last purchased.
+ * @param {Date} updatedData.dateNextPurchased The estimated date for the next purchase.
+ * @param {number} updatedData.totalPurchases The total number of times the item has been purchased.
+ * @returns {Promise<string>} A message confirming the item was successfully updated.
+ * @throws {Error} If the item update fails.
+ */
 export async function updateItem(
 	listPath,
 	itemId,
-	{ dateLastPurchased, totalPurchases },
+	{ dateLastPurchased, dateNextPurchased, totalPurchases },
 ) {
 	// reference the item path
 	const itemDocRef = doc(db, listPath, 'items', itemId);
 	// update the item with the purchase date and increment the total purchases made
 	try {
 		await updateDoc(itemDocRef, {
 			dateLastPurchased,
+			dateNextPurchased,
 			totalPurchases,
 		});
 		return 'item purchased';
-	} catch {
-		return;
+	} catch (error) {
```

Comment:
You should double check docs on what an error response looks like from Firebase. In the docs I was able to see that [`updateDoc`](https://firebase.google.com/docs/reference/js/firestore_lite.md#updatedoc_51a65e3) can return a [`FirestoreException`](https://firebase.google.com/docs/reference/js/firestore_lite.firestoreerror.md#firestoreerror_class). This error type is an object that has a `code`, `message`, and `stack` properties. If an error were to be caught by this try catch block, this error message would just record `Failed updating item: [Object]`, which would be unhelpful to the user. I'd suggest checking if `error.message` exists, and throwing that as part of the error.